### DISCUSSION TITLE
What is the cause of this error, knowing that it is running on another PC?

### DISCUSSION THREAD

**Discussion by mostafa770559** [community/community#77104] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

import cups
import pandas as pd
import matplotlib
from pylab import title, figure, xlabel, ylabel, xticks, bar, legend, axis, savefig
from fpdf import FPDF
import os
import time

import psutil as psutil

class CreatePdf:

    def __init__(self,Ar1,Ar2,Ar3,A):
        self.Ar1 = Ar1
        self.Ar2 = Ar2
        self.Ar3 = Ar3
        self.A=A
    def printPdf(self):
        os.startfile(self.A + '.pdf', "print")
        os.popen("Rundll32.exe SHELL32.dll, OpenAs_RunDLL "+self.A+'.pdf')
        time.sleep(5)
        for p in psutil.process_iter():  # Close Acrobat after printing the PDF
            if 'AcroRd' in str(p):
                p.kill()

 In another project

from CreatePdf import *

ass=CreatePdf(['a','s','d'],[12,3,4],[5,6,8],'sa')

ass.createPdf()
ass.printPd

**Comment by mostafa770559** [comment]
Traceback (most recent call last):
  File "C:\Users\drkha\PycharmProjects\pythonProject37\main.py", line 8, in <module>
    ass.printPdf()
  File "C:\Users\drkha\PycharmProjects\pythonProject37\CreatePdf.py", line 22, in printPdf
    os.startfile(self.A + '.pdf', "print")
OSError: [WinError 1155] No application is associated with the specified file for this operation: 'sa.pdf'

**Reply by MadScie254** [reply, marked as answer]
The error message you encountered indicates that there is no application associated with the specified file for the "print" operation. This means that the operating system on the PC where the code is running does not have a default program set to handle the printing of PDF files.

To resolve this issue, you can try the following steps:

1. Check if a PDF reader or printer software is installed on the PC. If not, install a PDF reader such as Adobe Acrobat Reader or a printer software that supports PDF printing.

2. Set the installed PDF reader or printer software as the default program for handling PDF files. You can usually do this by right-clicking on a PDF file, selecting "Open With," and choosing the desired application. Make sure to check the option to set it as the default program.

3. Restart the Python script and try printing the PDF file again. The operating system should now recognize the associated application and perform the print operation successfully.

If the issue persists, it might be necessary to review the code and ensure that the file path and name are correct, and that the PDF file has been generated successfully before attempting to print it.

**Comment by MadScie254** [comment]
The error message you encountered indicates that there is no application associated with the specified file for the "print" operation. This means that the operating system on the PC where the code is running does not have a default program set to handle the printing of PDF files.

To resolve this issue, you can try the following steps:

1. Check if a PDF reader or printer software is installed on the PC. If not, install a PDF reader such as Adobe Acrobat Reader or a printer software that supports PDF printing.

2. Set the installed PDF reader or printer software as the default program for handling PDF files. You can usually do this by right-clicking on a PDF file, selecting "Open With," and choosing the desired application. Make sure to check the option to set it as the default program.

3. Restart the Python script and try printing the PDF file again. The operating system should now recognize the associated application and perform the print operation successfully.

If the issue persists, it might be necessary to review the code and ensure that the file path and name are correct, and that the PDF file has been generated successfully before attempting to print it.
